script/amazon-linux-2023/setup-amazon-linux-2023 copy.sh

```
@@ -0,0 +1,333 @@
+#=========================================
+install_nodejs() {
+    check_command node && { log "NodeJS is already installed"; return 0; }
+
+    log "Installing NodeJS..."
+    curl -fsSL https://rpm.nodesource.com/setup_20.x | bash -
+    dnf install -y nodejs
+    npm install -g npm@latest
+    sudo npm install -g aws-cdk
+
+    # Add npm global bin to PATH and create symlink for cdk
+    local npm_global_bin=$(npm config get prefix)/bin
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)